drivers/nec/display/all.cr

```diff
@@ -0,0 +1,410 @@
+require "driver/interface/powerable"
+require "driver/interface/muteable"
+require "driver/interface/switchable"
+
+module Nec; end
+module Nec::Display; end
+
+# :title:All NEC Control Module
+#
+# Controls all LCD displays as of 1/07/2011
+# Status information avaliable:
+# -----------------------------
+#
+# (built in)
+# connected
+#
+# (module defined)
+# power
+# warming
+#
+# volume
+# volume_min == 0
+# volume_max
+#
+# brightness
+# brightness_min == 0
+# brightness_max
+#
+# contrast
+# contrast_min = 0
+# contrast_max
+#
+# audio_mute
+#
+# input (video input)
+# audio (audio input)
+
+class Nec::Display::All < PlaceOS::Driver
+  include Interface::Powerable
+  include Interface::AudioMuteable
+
+  enum Inputs
+    Vga          = 1
+    Rgbhv        = 2
+    Dvi          = 3
+    Hdmi_set     = 4
+    Video1       = 5
+    Video2       = 6
+    Svideo       = 7
+    Tuner        = 9
+    Tv           = 10
+    Dvd1         = 12
+    Option       = 13
+    Dvd2         = 14
+    Display_port = 15
+    Hdmi         = 17
+    Hdmi2        = 18
+    Hdmi3        = 130
+    Usb          = 135
+  end
+  include PlaceOS::Driver::Interface::InputSelection(Inputs)
+
+  # Discovery Information
+  tcp_port 7142
+  descriptive_name "NEC LCD Display"
+  generic_name :Display
+
+  default_settings({
+    volume_min: 0,
+    volume_max: 100
+  })
+
+  @volume_min : Int32 = 0
+  @volume_max : Int32 = 100
+
+  # 0x0D (<CR> carriage return \r)
+  DELIMITER = 0x0D_u8
+
+  @target_input : Input? = nil
+  @target_audio : Audio? = nil
+  @input_double_check : PlaceOS::Driver::Proxy::Scheduler? = nil
+
+  def on_load
+    # Communication settings
+    queue.delay = 120.milliseconds
+    transport.tokenizer = Tokenizer.new(Bytes[DELIMITER])
+    on_update
+  end
+
+  def on_update
+    @volume_min = setting(Int32, :volume_min)
+    @volume_max = setting(Int32, :volume_max)
+  end
+
+  def connected
+    schedule.every(50.seconds, true) do
+      do_poll
+    end
+  end
+
+  def disconnected
+    # Disconnected may be called without calling connected
+    # Hence the check if timer is nil here
```

Comment:
Again, any comments like this that do no contain information important to the operation of this specific driver can be dropped.